### PR TITLE
Audio types have the same naming convention as other SDL endian types, e.g. [S|U][BITS][LE|BE]

### PR DESCRIPTION
**Pull request by slouken**
Native endian types have no LE/BE suffix

### CODE REVIEW

**Comment by slouken** [comment]
FYI, in the process of working on this change, I found a few other places where we were using SDL_AUDIO_F32 when it should have been SDL_AUDIO_F32SYS, so this seems like a good change overall. The migration process will map the old SDL2 names as little endian explicitly, so that also seems good.

File: include/SDL3/SDL_audio.h [hunk 1]
```diff
@@ -80,15 +80,17 @@ typedef Uint16 SDL_AudioFormat;
 /* @{ */
 
 #define SDL_AUDIO_MASK_BITSIZE       (0xFF)
-#define SDL_AUDIO_MASK_DATATYPE      (1<<8)
-#define SDL_AUDIO_MASK_ENDIAN        (1<<12)
+#define SDL_AUDIO_MASK_FLOAT         (1<<8)
+#define SDL_AUDIO_MASK_LIL_ENDIAN    (1<<12)
+#define SDL_AUDIO_MASK_BIG_ENDIAN    (1<<13)
```
Comment (0x1F9F1) [inline include/SDL3/SDL_audio.h:85]: It feels a bit strange having separate bits for lil/big when they should always be mutually exclusive.

Comment (slouken) [reply include/SDL3/SDL_audio.h:85]: Yeah, I kind of agree, but then I found the code in SDL_audiocvt.c that strips off the mask to find the endian independent format.

Comment (0x1F9F1) [reply include/SDL3/SDL_audio.h:85]: I suppose so, but I'd say that's more of an edge case that isn't worth changing the format for.

Also, the old MSB formats now equal the new LE formats, which is quite confusing.

**Comment by sezero** [comment]
Sigh.......

Will you update sdl2-compat and SDL_mixer after this?

File: include/SDL3/SDL_audio.h (hunk 1 above)
Comment (slouken) [reply include/SDL3/SDL_audio.h:85]: Okay, I'll change that part back, thanks for the feedback!

**Comment by slouken** [comment]
> Sigh.......
> 
> Will you update sdl2-compat and SDL_mixer after this?

Yep :)

**Pull request merged**